seatunnel-connectors-v2/connector-hudi/src/main/java/org/apache/seatunnel/connectors/seatunnel/hudi/config/HudiSinkConfig.java

```diff
@@ -0,0 +1,110 @@
+ * limitations under the License.
+ */
+
+package org.apache.seatunnel.connectors.seatunnel.hudi.config;
+
+import org.apache.seatunnel.api.configuration.ReadonlyConfig;
+
+import org.apache.hudi.common.model.HoodieTableType;
+import org.apache.hudi.common.model.WriteOperationType;
+
+import lombok.Builder;
+import lombok.Data;
+
+import java.io.Serializable;
+import java.util.Optional;
+
+@Data
+@Builder(builderClassName = "Builder")
+public class HudiSinkConfig implements Serializable {
+
+    private static final long serialVersionUID = 2L;
+
+    private String tableName;
+
+    private String tablePath;
+
+    private int insertShuffleParallelism;
+
+    private int upsertShuffleParallelism;
+
+    private int minCommitsToKeep;
+
+    private int maxCommitsToKeep;
+
+    private HoodieTableType tableType;
+
+    private WriteOperationType opType;
+
+    private String confFile;
+
+    private int batchIntervalMs;
+
+    private String recordKeyFields;
+
+    private String partitionFields;
+
+    public static HudiSinkConfig of(ReadonlyConfig config) {
+        HudiSinkConfig.Builder builder = HudiSinkConfig.builder();
+        Optional<Integer> optionalInsertShuffleParallelism =
+                config.getOptional(HudiOptions.INSERT_SHUFFLE_PARALLELISM);
+        Optional<Integer> optionalUpsertShuffleParallelism =
+                config.getOptional(HudiOptions.UPSERT_SHUFFLE_PARALLELISM);
+        Optional<Integer> optionalMinCommitsToKeep =
+                config.getOptional(HudiOptions.MIN_COMMITS_TO_KEEP);
+        Optional<Integer> optionalMaxCommitsToKeep =
+                config.getOptional(HudiOptions.MAX_COMMITS_TO_KEEP);
+
+        Optional<HoodieTableType> optionalTableType = config.getOptional(HudiOptions.TABLE_TYPE);
+        Optional<WriteOperationType> optionalOpType = config.getOptional(HudiOptions.OP_TYPE);
+
+        Optional<Integer> optionalBatchIntervalMs =
+                config.getOptional(HudiOptions.BATCH_INTERVAL_MS);
+
+        Optional<String> partitionFields = config.getOptional(HudiOptions.PARTITION_FIELDS);
+
+        Optional<String> recordKeyFields = config.getOptional(HudiOptions.RECORD_KEY_FIELDS);
+
+        builder.confFile(config.get(HudiOptions.CONF_FILES));
+        builder.tableName(config.get(HudiOptions.TABLE_NAME));
+        builder.tablePath(config.get(HudiOptions.TABLE_PATH));
+        builder.tableType(optionalTableType.orElseGet(HudiOptions.TABLE_TYPE::defaultValue));
+        builder.opType(optionalOpType.orElseGet(HudiOptions.OP_TYPE::defaultValue));
+
+        builder.batchIntervalMs(
+                optionalBatchIntervalMs.orElseGet(HudiOptions.BATCH_INTERVAL_MS::defaultValue));
+
+        builder.partitionFields(
+                partitionFields.orElseGet(HudiOptions.PARTITION_FIELDS::defaultValue));
+
+        builder.recordKeyFields(
+                recordKeyFields.orElseGet(HudiOptions.RECORD_KEY_FIELDS::defaultValue));
+
+        builder.insertShuffleParallelism(
```

Comment:
why not use `builder.insertShuffleParallelism(config.get(HudiOptions.UPSERT_SHUFFLE_PARALLELISM))` ?